public/pages/CreateSnapshotPolicy/components/CronSchedule/CronSchedule.tsx

```diff
@@ -0,0 +1,199 @@
+    // changeCron({ dayOfWeek });
+  }
+
+  function onDayOfMonthChange(dayOfMonth: number) {
+    setMonth(dayOfMonth);
+    // changeCron({ dayOfMonth });
+  }
+
+  function onStartTimeChange(date: moment.Moment) {
+    const minute = date.minute();
+    const hour = date.hour();
+    setMinute(minute);
+    setHour(hour);
+    // changeCron({ minute, hour });
+  }
+
+  function onTypeChange(e: ChangeEvent<HTMLSelectElement>) {
+    const frequencyType = e.target.value;
+    onChangeFrequencyType(e);
+    changeCron({ frequencyType });
+  }
+
+  const dayOfWeekCheckbox = (day: string, checkedDay: string) => (
+    <EuiFlexItem key={day} grow={false} style={{ marginRight: "0px" }}>
+      <EuiCheckbox id={day} label={day} checked={checkedDay === day} onChange={(e) => onDayOfWeekChange(day)} compressed />
+    </EuiFlexItem>
+  );
+
+  let additionalContent;
+  if (frequencyType === "weekly") {
+    // TODO SM if use dayOfWeek not initWeek, somehow it would be SUN
+    additionalContent = <EuiFlexGroup>{WEEK_DAYS.map((d) => dayOfWeekCheckbox(d, initWeek))}</EuiFlexGroup>;
+  }
+  if (frequencyType === "monthly") {
+    additionalContent = (
+      <>
+        <CustomLabel title="On the" />
+        <EuiFlexGroup gutterSize="m">
+          <EuiFlexItem>
+            <EuiSelect options={[{ value: "day", text: "Day" }]} defaultValue="Day" />
+          </EuiFlexItem>
+          <EuiFlexItem>
+            <EuiFieldNumber
+              value={dayOfMonth}
+              onChange={(e) => {
+                console.log(`sm dev change day of month ${parseInt(e.target.value)}`);
+                onDayOfMonthChange(parseInt(e.target.value));
+              }}
+            />
+          </EuiFlexItem>
+        </EuiFlexGroup>
+      </>
+    );
+  }
+
+  const cronExpressionHelpText = (
+    <EuiText color="subdued" size="s" style={{ padding: "5px 0px" }}>
+      <p style={{ fontWeight: 200, fontSize: "12px" }}>
+        Use Cron expression to define complex schedule.{" "}
+        <EuiLink href={CRON_EXPRESSION_DOCUMENTATION_URL} target="_blank">
+          Learn more
+        </EuiLink>
+      </p>
+    </EuiText>
+  );
+
+  return (
+    <>
+      <CustomLabel title="Schedule frequency" />
+      <EuiSelect id="creationCronScheduleType" options={CRON_SCHEDULE_FREQUENCY_TYPE} value={frequencyType} onChange={onTypeChange} />
+
+      <EuiSpacer size="m" />
+
+      <EuiFlexGroup justifyContent="flexStart" alignItems="flexStart">
+        <EuiFlexItem style={{ maxWidth: 400 }}>
+          {frequencyType === "custom" ? (
+            <>
+              <CustomLabel title="Cron expression" helpText={cronExpressionHelpText} />
+              <EuiFieldText
+                value={cronExpression}
+                onChange={(e) => {
+                  onCronExpressionChange(e.target.value);
+                }}
+              />
+            </>
+          ) : (
+            <>
+              <CustomLabel title="Start time" />
+              <EuiDatePicker
```

Comment:
As discussed offline, this element should likely be an `EuiFieldNumber` or something similar when the `Schedule frequency` selection is `Hourly`. In that case, we want the user to be able to select a number between 0 and 59 which represents the minute of the hour they want to take the snapshot.

Reply:
Good suggestion! When user choose the hourly frequency, we will default to start of the hour. If user want to choose specific minute, they can use cron expression.